os/kernel/debug/mem_leak_checker.c

```diff
@@ -52,27 +52,37 @@ struct alloc_node_info_s {
 	struct alloc_node_info_s *next;
 };
 
-static struct alloc_node_info_s *hash_table[HASH_SIZE];
-static struct alloc_node_info_s node_info[MAX_ALLOC_COUNT];
-static struct mem_leak_checker_info_s leak_checker;
-static int checkerpid;
-
-#ifdef CONFIG_MM_KERNEL_HEAP
-extern struct mm_heap_s g_kmmheap[CONFIG_KMM_NHEAPS];
-#endif
+static struct alloc_node_info_s **hash_table;
+static struct alloc_node_info_s *node_info;
 
 static void hash_init(void)
 {
 	int index;
+
+	DEBUGASSERT(hash_table == NULL && node_info == NULL);
+
+	hash_table = (struct alloc_node_info_s **)malloc(sizeof(struct alloc_node_info_s *) * HASH_SIZE);
```

Comment:
It alloc properly for all the time?

Reply:
Thank you for review.
I added error handling

Reply:
Thank you for review.
I added error handling